pkgpanda/util.py

```diff
@@ -150,6 +151,38 @@ def get_requests_retry_session(max_retries=4, backoff_factor=1, status_forcelist
     return session
 
 
+def stream_remote_file_with_retries(out_filename, url, retries=4):
+    def cleanup():
+        try:
+            os.remove(out_filename)
+        except Exception:
+            pass
+
+    while True:
+        with open(out_filename, "w+b") as f:
+            r = get_requests_retry_session().get(url, stream=True)
+            if r.status_code == 301:
+                raise Exception("got a 301")
+            r.raise_for_status()
+
+            for chunk in r.iter_content(chunk_size=4096):
+                f.write(chunk)
+            f.flush()
```

Comment:
Consider maintaining "total bytes read" by accumulating `len(chunk)` instead of `stat`ing the file and explicitly flushing it.

Reply:
That's a good idea!